src/server/tx_base.cc

```diff
@@ -18,8 +18,7 @@ using Payload = journal::Entry::Payload;
 void RecordJournal(const OpArgs& op_args, string_view cmd, ArgSlice args, uint32_t shard_cnt,
                    bool multi_commands) {
   VLOG(2) << "Logging command " << cmd << " from txn " << op_args.tx->txid();
-  op_args.tx->LogJournalOnShard(op_args.shard, Payload(cmd, args), shard_cnt, multi_commands,
-                                false);
+  op_args.tx->LogJournalOnShard(op_args.shard, Payload(cmd, args), shard_cnt, multi_commands, true);
```

Comment:
The call to RecordJournal is always from non auto journal commands where we call RecordJournal inside the command callback. Untill now we did not allow preemption in callbacks, we have some assumptions in code that callbacks do not preempt. We do need to fix them as part of the effort of Big values serialization Shahar is working on

Reply:
I agree with @dranikpg  that a comment is missing that explains why we need to avoid preemption.
Also, I am SURE it is possible to fix the logic so that we won't aggregate gigabytes of memory even if we do not allow preemption in the callback.

Reply:
Yes I agree. I remember that this fix - https://github.com/dragonflydb/dragonfly/pull/3016/files should fix the problem. I will check the issue again